src/lib/core/Controller.tsx

```diff
@@ -77,6 +110,16 @@ function injectControllerBehavior(proto: React.Component) {
       }
     })
   })
+
+  patchReturnMethod(ComponentClass.prototype, 'render', function(this: Controller, prev?: React.ReactElement<{}> | null) {
+    const { shouldRender } = this['@subtreeLoader']
+    return shouldRender ? prev || null : null
```

Comment:
Is this the same as `shouldRender && prev || null`?  Might be nicer as if/else logic anyway as it's quite hard to see what's going on at a glance.